spec/mainadditions.html

```diff
@@ -188,6 +188,40 @@ <h1>Overview of Date Objects and Definitions of Abstract Operations</h1>
 
     <p>[...]</p>
 
+    <emu-clause id="sec-temporal-getutcepochnanoseconds" type="abstract operation">
+      <h1>
+        GetUTCEpochNanoseconds (
+          _year_: an integer,
+          _month_: an integer in the inclusive interval from 1 to 12,
+          _day_: an integer in the inclusive interval from 1 to 31,
+          _hour_: an integer in the inclusive interval from 0 to 23,
+          _minute_: an integer in the inclusive interval from 0 to 59,
+          _second_: an integer in the inclusive interval from 0 to 59,
+          _millisecond_: an integer in the inclusive interval from 0 to 999,
+          _microsecond_: an integer in the inclusive interval from 0 to 999,
+          _nanosecond_: an integer in the inclusive interval from 0 to 999,
+          <ins>optional _offsetNanoseconds_: an integer in the exclusive interval from -nsPerDay to nsPerDay,</ins>
+        ): a BigInt
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>
+          The returned value represents a number of nanoseconds since the epoch that corresponds to the given ISO 8601 calendar date and wall-clock time in UTC.
+          <ins>If a UTC offset is provided, the returned value will be adjusted by that offset.</ins>
+        </dd>
+      </dl>
+      <emu-alg>
+        1. Let _date_ be MakeDay(𝔽(_year_), 𝔽(_month_ - 1), 𝔽(_day_)).
+        1. Let _time_ be MakeTime(𝔽(_hour_), 𝔽(_minute_), 𝔽(_second_), 𝔽(_millisecond_)).
+        1. Let _ms_ be MakeDate(_date_, _time_).
```

Comment:
Note, there is some interaction with https://github.com/tc39/proposal-temporal/pull/2518 here but I think it's OK not to consider that in scope for this PR because this definition comes from ECMA-262 already. It can be considered part of https://github.com/tc39/ecma262/issues/1087